File: Stack/stack.py
class stack:
	"""
	Stack implementation using lists
	"""
	def __init__(self):
		self.lst = []
		self.top = None

	def insert(self, value):
		self.lst.append(value)
		self.top = self.lst[-1]

	def pop(self):
		self.top = self.lst[-2]
		return self.lst.pop()

	def is_empty(self):
		if len(self.lst) >= 1:
			return False
		else:
			return True

	def peek(self):
		return self.top

	def print_stack(self):
		for i in self.lst[-1::-1]:
			print(i)

x = stack()
for i in range(10):
	x.insert(i)
x.print_stack()

File: Stack/stack.c
#include <stdio.h>
#include <stdlib.h>


#define m 5
int st[m];
int top = -1;


void push(int x)
{
    if (top >= m - 1)
    {
        printf("Stack overflow\n");
    }
    else
    {
        top = top + 1;
        st[top] = x;
        printf("%d pushed\n", x);
    }
}


int pop()
{
    if (top == -1)
    {
        printf("Stack underflow\n");
    }
    else
    {
        printf("%d popped\n", st[top]);
        top = top - 1;
        return st[top + 1];
    }   
}


int peek()
{
    if (top >= 0)
    {
        return st[top];
    }
    else
    {
        printf("Nothing here to peek\n");
    }
    
}


int main()
{
    push(1);
    push(2);
    push(3);
    push(4);
    push(5);
    push(6);
    pop();
    pop();
    pop();
    pop();
    peek();
    pop();
    pop();
    pop();
    pop();
    return 0;
}